cmd/edge-orchestration/main.go

```diff
@@ -190,6 +193,12 @@ func orchestrationInit() error {
 			go discoverymgr.GetInstance().StartMNEDCClient(deviceIDFilePath, mnedcServerConfig)
 		}
 	}
+	if len(cloudsync) > 0 {
+		//initiate cloudsync
+		cloudsyncmgr.GetInstance().StartCloudSync(cloudsync)
+	} else {
+		log.Info(logPrefix, "Clousync is off")
+	}
```

Comment:
This is a similar problem to #178. Please move these code to [orchestrationapi](https://github.com/lf-edge/edge-home-orchestration-go/blob/9c738811ac353669c73a33b5f6a5d0d330090c7a/internal/orchestrationapi/orchestration.go#L201). See #195 for details.

Reply:
Why should the cloudsync started by discovery manager? and this code is in uniformity with the code to check if mnedc is set in env varibale in same main file

Reply:
Please take a closer look at orchestrationapi package. The orchestrationapi package must run controllers. MNEDC in main.go is an issue since it's not implemented according to the architecture.

Reply:
@nitu-s-gupta Thank you for the diligent analysis and discussion. I personally think that we all would be better to take a closer look on our previous discussion at #178. Theoretically `cloudsync` seems to be independent to `discovery manager`, but in our architecture, it is all interconnected to `orchestrationapi` and our `MNEDC` implementation should be revised and amended w.r.t. our architecture from `orchestrationapi` perspective.

Reply:
And this is the basis of argument here. 😁

Reply:
Taking a closer look at main.go I could find 
isSecured := false
	if len(secure) > 0 {
		if strings.Compare(strings.ToLower(secure), "true") == 0 {
			log.Println("Orchestration init with secure option")
			isSecured = true
		}
	}

	cipher := dummy.GetCipher(cipherKeyFilePath)
	if isSecured {
		verifier.Init(containerWhiteListPath)
		authenticator.Init(passPhraseJWTPath)
		authorizer.Init(rbacRulePath)
		cipher = sha256.GetCipher(cipherKeyFilePath)
	}
seems this initiation also need to be moved?

Reply:
> Taking a closer look at main.go I could find isSecured := false if len(secure) > 0 { if strings.Compare(strings.ToLower(secure), "true") == 0 { log.Println("Orchestration init with secure option") isSecured = true } }
> 
> ```
> cipher := dummy.GetCipher(cipherKeyFilePath)
> if isSecured {
> 	verifier.Init(containerWhiteListPath)
> 	authenticator.Init(passPhraseJWTPath)
> 	authorizer.Init(rbacRulePath)
> 	cipher = sha256.GetCipher(cipherKeyFilePath)
> }
> ```
> 
> seems this initiation also need to be moved?

Is this controller?

Reply:
When i say this i mean the variable check that are done which are set by env, i think there seems to be some confusion

Reply:
@tdrozdovsky What do you think 

> Taking a closer look at main.go I could find isSecured := false if len(secure) > 0 { if strings.Compare(strings.ToLower(secure), "true") == 0 { log.Println("Orchestration init with secure option") isSecured = true } }
> 
> ```
> cipher := dummy.GetCipher(cipherKeyFilePath)
> if isSecured {
> 	verifier.Init(containerWhiteListPath)
> 	authenticator.Init(passPhraseJWTPath)
> 	authorizer.Init(rbacRulePath)
> 	cipher = sha256.GetCipher(cipherKeyFilePath)
> }
> ```
> 
> seems this initiation also need to be moved? Anyways though I will move cloudsync part, but it was this code lines uniformity that i could find at multiple places

@tdrozdovsky What do you think about moving this code to securemgr and orchestrationapi? I think we can talk about the securemgr refactoring.

Reply:
> When i say this i mean the variable check that are done which are set by env, i think there seems to be some confusion

I don't really care much about checking env variables in main.go, however I believe orchestrationapi package should orchestrates other controllers(modules).
Also, it is recommended not to put a lot of business logic code in the cmd folder. Please see [golang standard guide](https://github.com/golang-standards/project-layout/tree/master/cmd).

Reply:
@nitu-s-gupta Let us take a step one by one. Let us focus on the diff. from this PR. The point is to follow the architecture principle of `orchestrationapi` and we need to revise this PR for the required clearance. In other words, this is in the end to resolve the issue #178. 🙏

Reply:
Then as per our discussion what i get is two options
1. Check the env vairable in the orchestration.go file Start function (line 201) and then start the instance
2. moving the above lines of code to check evn variable and then start syncmanager in the respective controller  initialization part and start the synmgr instance from Start function line 201 of orchestration.go file
I would prefer option 2 to maintian the uniformity of file. Pelase let me know if we are on same lines or still there is some confusion

Reply:
Can you suggest code with option 2? It seems good to me and we can talk more about with your updated PR.

Reply:
Updated the PR, where the controller checks for value of host thereby moving the respective check to the respective controller and orchestrationapi starts the cloudsync manager as mentioned in our discussion. For me seems to resolve the architecture issue